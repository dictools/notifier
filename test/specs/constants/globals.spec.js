import GLOBALS from "@constants/globals";

describe("<root>/constants/globals.js", () => {
  test("Should globals match snapshot", async () => {
    expect(GLOBALS).toMatchSnapshot();
  });
});
