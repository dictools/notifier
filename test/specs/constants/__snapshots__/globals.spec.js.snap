// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<root>/constants/globals.js Should globals match snapshot 1`] = `
Object {
  "COLORS": Object {
    "ADDED": "#178667",
    "CHANGED": "#FEC621",
    "REMOVED": "#D73645",
  },
  "CORE": Object {
    "TASK": "<#{url}/#{task}|#{task}>",
  },
  "MESSAGES": Object {
    "ADDED": Object {
      "BODY": "*Text:* #{value}",
      "DESCRIPTION": "Dictionary key *#{key}* has been #{type}",
    },
    "CHANGED": Object {
      "BODY": "*From:* #{oldValue}
*To:* #{value}",
      "DESCRIPTION": "Dictionary key *#{key}* has been #{type}",
    },
    "FOOTER": "*#{type} by:* #{username}",
    "FOOTER_WITH_TASK": "*#{type} by:* #{username} | *Task:* #{task}",
    "HOOK_ERROR": "
[31m[Notifier] Error (#{code}): #{message}[0m
",
    "HOOK_NOTIFIED": "
[32m[Notifier] Changes were notified.[0m
",
    "KEYS_UP_TO_DATE": "
[32m[Notifier] Dictionary keys are up to date.[0m
",
    "REMOVED": Object {
      "BODY": "*Text:* #{value}",
      "DESCRIPTION": "Dictionary key *#{key}* has been #{type}",
    },
    "TITLE": "_Changes to #{project} dictionary key(s):_

",
    "VALIDATIONS": Object {
      "hookUri": "Please define a hookUri to post the report { hookUri: <String> }",
      "path": "Please define a dictionary path: { path: <String> }",
      "project": "Please define the project properties { project: <Object> }",
      "project.boardUrl": "Please define a project board url { project.boardUrl: <String> }",
      "project.key": "Please define a project key { project.key: <String|Array|Regex> }",
      "project.name": "Please define a project name { project.name: <String> }",
    },
  },
}
`;
